doc/source/examples/mountain.cpp

```diff
@@ -21,10 +21,10 @@ main()
     // raster grid and boundary conditions
     fs::raster_boundary_status bs(fs::node_status::fixed_value);
 
-    auto grid = fs::raster_grid::from_length({ 201, 301 }, { 5e4, 7.5e4 }, bs);
+    auto grid = fs::raster_grid<>::from_length({ 201, 301 }, { 5e4, 7.5e4 }, bs);
```

Comment:
```suggestion
    auto grid = fs::raster_grid::from_length({ 201, 301 }, { 5e4, 7.5e4 }, bs);
```

Let's use C++17 for the documentation snippets. Same suggestion for all other relevant places in the examples and user guides.

Not sure for the tests and benchmarks, though. Any reason we don't want to force C++17 for those?

Reply:
`fs::raster_grid::from_length` is invalid C++17 CTAD
see https://en.cppreference.com/w/cpp/language/class_template_argument_deduction

Reply:
Ah ok, not a big deal.